orm/src/test/java/org/hibernate/search/test/util/ResourceCleanupFunctions.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.test.util;
+
+import java.util.function.Consumer;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+
+/**
+ * Some helpers to allow using lamda blocks to ensure that
+ * resources which get opened are also being closed correctly.
+ *
+ * @author Sanne Grinovero (C) 2016 Red Hat Inc.
+ */
+public final class ResourceCleanupFunctions {
+
+	private ResourceCleanupFunctions() {
+		//Utility class: not to be constructed.
+	}
+
+	/**
+	 * Given an EntityManagerFactory, we'll open an EntityManager, allow some
+	 * code block to use it as a FullTextEntityManager, and finally close
+	 * the EntityManager.
+	 * @param emf The EntityManagerFactory which we'll use to create a new EntityManager
+	 * @param consumer A block of code which uses the FullTextEntityManager
+	 */
+	public static void withinEntityManager(EntityManagerFactory emf, Consumer<FullTextEntityManager> consumer) {
+		EntityManager entityManager = emf.createEntityManager();
+		try {
+			FullTextEntityManager fem = Search.getFullTextEntityManager( entityManager );
+			consumer.accept( fem );
+		}
+		finally {
+			entityManager.close();
+		}
```

Comment:
Maybe in that case... Though I'm not sure that exceptions get logged when they are bypassed due to another exception being thrown in a finally block.
Before Java 7, people advised to explicitly log those, but it might be because the default behavior simply sends the stacktrace to `System.err`.
Since Java 7, I must admit I don't know what happens in that case.

In the helper below, though, I'm positive the Throwable t gets swallowed and never appears in logs if an exception arises when rolling back.

Sorry to be pestering, but exceptions being swallowed in tests, especially assertion failures, are a real pain. And since we get to solve the issue in a single, widely used helper, I don't see why we wouldn't...